### PR TITLE
utilise py-jail instead of a cached jail state

### PR DESCRIPTION
**Pull request by gronke**
- replaces fork of `/usr/sbin/jls` to query a jails JID
- [py-jail](https://github.com/gronke/py-jail) is a libc binding for jail interaction using native Python3
- Always queries the jail live with libc.jail_get - no more need for manual state querying

- breaks ioc CLI commans that expect the jail.state object

### CODE REVIEW

**Comment by gronke** [comment]
@himrock922 with this you no longer have to call `jail.state.query()` manually!

Again, thanks to @fabianfreyer, who brought the inspiration this step with [libjail-rs](https://github.com/fubarnetes/libjail-rs/).

File: libioc/Jail.py [hunk 1]
```diff
@@ -1640,7 +1640,7 @@ def _launch_args(self) -> typing.List[str]:
                     # vnet is only used when explicitly enabled
                     # (friendly to Kernels without VIMAGE support)
                     continue
-                value = "vnet"
+                value = 1
```
Comment (igalic) [inline libioc/Jail.py]: oops

**Comment by gronke** [comment]
The features used by libioc are now [unit-tested in py-jail](https://github.com/gronke/py-jail/commit/251a3fc2e2421189339e1ce93dffc0581b7df3c1).

File: tests/conftest.py [hunk 2]
```diff
@@ -175,6 +178,49 @@ def release(
...
+    del release
+
+
+@pytest.fixture(scope="function")
+def new_jail(
+    host: 'libioc.Host.Host',
+    logger: 'libioc.Logger.Logger',
+    zfs: libzfs.ZFS,
+) -> 'libioc.Jail.Jail':
+    jail_name = "new-jail-" + str(random.randint(1, 32768))
```
Comment (igalic) [inline tests/conftest.py]: this might be better done with Hypothesis

File: tests/conftest.py [hunk 3]
```diff
@@ -175,6 +178,49 @@ def release(
...
+        logger=logger,
+        zfs=zfs
+    )
+    yield new_jail
+    new_jail.destroy(force=True)
+
+
+@pytest.fixture(scope="function")
+def bridge_interface() -> str:
+    bridge_name = "bridgeTest" + str(random.randint(1024, 4096))
```
Comment (igalic) [inline tests/conftest.py]: again, this might be better done with Hypothesis: https://hypothesis.readthedocs.io/en/latest/data.html

**Pull request merged**